File: llm.py
import dashscope
from dashscope import Generation

# 设置你的 API Key（建议用环境变量）
dashscope.api_key = "YOUR_DASHSCOPE_API_KEY"

def call_qwen(prompt: str):
    try:
        response = Generation.call(
            model="qwen-plus",  # 可选: qwen-turbo, qwen-max, qwen-plus
            prompt=prompt,
            max_tokens=500,
            temperature=0.7
        )
        if response.status_code == 200:
            return response.output.text.strip()
        else:
            return f"❌ 请求失败: {response.code} - {response.message}"
    except Exception as e:
        return f"❌ 异常: {str(e)}"

# 使用示例
if __name__ == "__main__":
    user_input = "请用一句话解释什么是人工智能？"
    answer = call_qwen(user_input)
    print("👤 用户:", user_input)
    print("🤖 Qwen:", answer)

File: llm.java
import okhttp3.*;
import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;

import java.io.IOException;

public class QwenClient {

    private static final String API_URL = "https://dashscope.aliyuncs.com/api/v1/services/aigc/text-generation/generation";
    private static final String API_KEY = "YOUR_DASHSCOPE_API_KEY"; // 替换为你的 Key
    private static final OkHttpClient client = new OkHttpClient();
    private static final ObjectMapper mapper = new ObjectMapper();

    public static String callQwen(String prompt) {
        try {
            // 构建请求体
            String jsonBody = "{\n" +
                "  \"model\": \"qwen-plus\",\n" +
                "  \"input\": {\n" +
                "    \"messages\": [\n" +
                "      {\"role\": \"user\", \"content\": \"" + prompt.replace("\"", "\\\"") + "\"}\n" +
                "    ]\n" +
                "  },\n" +
                "  \"parameters\": {\n" +
                "    \"max_tokens\": 500,\n" +
                "    \"temperature\": 0.7\n" +
                "  }\n" +
                "}";

            Request request = new Request.Builder()
                .url(API_URL)
                .post(RequestBody.create(jsonBody, MediaType.get("application/json; charset=utf-8")))
                .addHeader("Authorization", "Bearer " + API_KEY)
                .addHeader("Content-Type", "application/json")
                .build();

            try (Response response = client.newCall(request).execute()) {
                if (!response.isSuccessful()) {
                    return "❌ HTTP 错误: " + response.code();
                }

                String responseBody = response.body().string();
                JsonNode root = mapper.readTree(responseBody);
                JsonNode output = root.path("output").path("text");

                if (output.isMissingNode()) {
                    return "❌ 解析失败: " + responseBody;
                }
                return output.asText().trim();
            }
        } catch (IOException e) {
            return "❌ 网络异常: " + e.getMessage();
        }
    }

    public static void main(String[] args) {
        String prompt = "请用一句话解释什么是人工智能？";
        String answer = callQwen(prompt);
        System.out.println("👤 用户: " + prompt);
        System.out.println("🤖 Qwen: " + answer);
    }
}